compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -269,8 +272,10 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
                     ? ASTBuilderUtil.createLiteral(pos, symTable.nilType, Names.NIL_VALUE)
                     : onConflictExpr;
             BLangVariableReference tableRef = addTableConstructor(queryExpr, queryBlock);
+            BLangLiteral isReadonly = ASTBuilderUtil.createLiteral(null, symTable.booleanType,
```

Comment:
Shall we use the query position here instead of passing null as the position

Reply:
Since this is kind of internal variable which is used to pass a flag to helper function, is that right to set the position of Query to this literal?

Reply:
It'll be okay I guess. `pos` has been used in some other places in the query desugar too. Passing null might lead to NPEs

Reply:
Ah okay. Will update. Thanks.